modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/discovery/impl/DiscoveryServiceImpl.java

```diff
@@ -61,35 +61,26 @@
      * @param response         - DiscoveryResponse bean for response
      */
     @Override
-    public void discover(DiscoveryRequest discoveryRequest, Holder<DiscoveryResponse> response) {
+    public void discover(DiscoveryRequest discoveryRequest, Holder<DiscoveryResponse> response)
+            throws WindowsDeviceEnrolmentException {
 
-        ServletContext ctx = (ServletContext) context.getMessageContext().get(MessageContext.SERVLET_CONTEXT);
-        WindowsPluginProperties windowsPluginProperties = (WindowsPluginProperties) ctx.getAttribute(
-                PluginConstants.WINDOWS_PLUGIN_PROPERTIES);
+        String emailId = discoveryRequest.getEmailId();
+        String[] domainUserArray = emailId.split(DELIMITER);
```

Comment:
You don't need to specify "Array" on the variable name above. You can keep it s "domainUsers" or something similar.